common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
+                return true;
+            }
+
+            // If we can already execute, there is nothing to do.
+            if (file.canExecute()) {
+                return true;
+            }
+
+            // On volumes, with noexec set, even files with the executable POSIX permissions will fail to execute.
+            // So, check for executable permissions, then find out if the file can be executed.
+
+            // Everything is done reflectively to prevent the animal-sniffer plugin from complainin.
+            Class<?> filesClass = Class.forName("java.nio.file.Files");
```

Comment:
@Scottmitch The non-reflective version is in, though it requires suppressing checks for uses of all of:

* java.nio.file.attribute.PosixFilePermission
* java.nio.file.Files
* java.nio.file.LinkOption
* java.nio.file.Path
* java.io.File (so it can call `File#toPath()`)

Note that the research in [this repository](https://github.com/mkw/java6test/tree/master) seems to have convinced @normanmaurer that the Java7 check at the beginning of the method should work (see the comments on #6822) in Java6 with the imports present.